mixer/pkg/il/compiler/compiler_test.go

```diff
@@ -15,1172 +15,50 @@
-		result: []byte(net.IPv4zero),
-		code: `fn eval() interface
-  apush_s "0.0.0.0"
-  call ip
-  ret
-end`,
-	},
-	{
-		expr: `aip == bip`,
-		input: map[string]interface{}{
-			"aip": []byte{0x1, 0x2, 0x3, 0x4},
-			"bip": []byte{0x4, 0x5, 0x6, 0x7},
-		},
-		result: false,
-		code: `fn eval() bool
-  resolve_f "aip"
-  resolve_f "bip"
-  call ip_equal
-  ret
-end`,
-	},
-	{
-		expr:   `timestamp("2015-01-02T15:04:35Z")`,
-		result: t,
-		code: `fn eval() interface
-  apush_s "2015-01-02T15:04:35Z"
-  call timestamp
-  ret
-end`,
-	},
-	{
-		expr: `t1 == t2`,
-		input: map[string]interface{}{
-			"t1": t,
-			"t2": t2,
-		},
-		result: false,
-		code: `fn eval() bool
-  resolve_f "t1"
-  resolve_f "t2"
-  call timestamp_equal
-  ret
-end`,
-	},
-	{
-		expr: `t1 == t2`,
-		input: map[string]interface{}{
-			"t1": t,
-			"t2": t,
-		},
-		result: true,
-		code: `fn eval() bool
-  resolve_f "t1"
-  resolve_f "t2"
-  call timestamp_equal
-  ret
-end`,
-	},
-	{
-		expr: `t1 != t2`,
-		input: map[string]interface{}{
-			"t1": t,
-			"t2": t,
-		},
-		result: false,
-		code: `fn eval() bool
-  resolve_f "t1"
-  resolve_f "t2"
-  call timestamp_equal
-  not
-  ret
-end`,
-	},
-	{
-		expr: `t1 | t2`,
-		input: map[string]interface{}{
-			"t2": t2,
-		},
-		result: t2,
-	},
-	{
-		expr: `t1 | t2`,
-		input: map[string]interface{}{
-			"t1": t,
-			"t2": t2,
-		},
-		result: t,
-	},
-}
-
-var globalConfig = pb.GlobalConfig{
-	Manifests: []*pb.AttributeManifest{
-		{
-			Attributes: map[string]*pb.AttributeManifest_AttributeInfo{
-				"ai": {
-					ValueType: pbv.INT64,
-				},
-				"ab": {
-					ValueType: pbv.BOOL,
-				},
-				"as": {
-					ValueType: pbv.STRING,
-				},
-				"ad": {
-					ValueType: pbv.DOUBLE,
-				},
-				"ar": {
-					ValueType: pbv.STRING_MAP,
-				},
-				"adur": {
-					ValueType: pbv.DURATION,
-				},
-				"at": {
-					ValueType: pbv.TIMESTAMP,
-				},
-				"aip": {
-					ValueType: pbv.IP_ADDRESS,
-				},
-				"bi": {
-					ValueType: pbv.INT64,
-				},
-				"bb": {
-					ValueType: pbv.BOOL,
-				},
-				"bs": {
-					ValueType: pbv.STRING,
-				},
-				"bd": {
-					ValueType: pbv.DOUBLE,
-				},
-				"br": {
-					ValueType: pbv.STRING_MAP,
-				},
-				"bdur": {
-					ValueType: pbv.DURATION,
-				},
-				"bt": {
-					ValueType: pbv.TIMESTAMP,
-				},
-				"t1": {
-					ValueType: pbv.TIMESTAMP,
-				},
-				"t2": {
-					ValueType: pbv.TIMESTAMP,
-				},
-				"bip": {
-					ValueType: pbv.IP_ADDRESS,
-				},
-				"b1": {
-					ValueType: pbv.BOOL,
-				},
-				"b2": {
-					ValueType: pbv.BOOL,
-				},
-				"sm": {
-					ValueType: pbv.STRING_MAP,
-				},
-			},
-		},
-	},
-}
-
 func TestCompile(t *testing.T) {
 
-	finder := descriptor.NewFinder(&globalConfig)
+	for i, test := range ilt.TestData {
+		if test.E == "" {
+			continue
+		}
 
-	for i, te := range tests {
-		t.Run(fmt.Sprintf("%d '%s'", i, te.expr), func(tt *testing.T) {
-			result, err := Compile(te.expr, finder)
+		conf := test.Conf
```

Comment:
Curious: is there a reason why below lines are not inside the `t.Run (...`. AFAIK, that will do nice parallelization for all tests inside the ilt.TestData
```
+		conf := test.Conf
 +		if conf == nil {
 +			conf = ilt.TestConfigs["Default"]
 +		}
 +		finder := descriptor.NewFinder(conf)
 +
 +		name := fmt.Sprintf("%d '%s'", i, test.E)
```

Reply:
Short answer: Nothing particular, moved code. Long Answer: I started with the idea of writing an AST version of this as well, but decided to consolidate all AST/IL comparison in evaluator instead.